crates/compiler/mono/src/tail_recursion.rs

```diff
@@ -403,20 +403,26 @@ fn insert_jumps<'a>(
     }
 }
 
+#[derive(Debug, Clone)]
+enum RecCallLocation<'a> {
+    DirectlyInTag,
+    MemberOfStruct(Symbol, &'a [u64]),
```

Comment:
this `&'a []` means we get more lifetimes in the code and need to pass the arena. Maybe it's better to also make this an arrayvec? very deep paths are unlikely, I'd think 4, or maybe 8 indices at most is fine. I think that simplifies the code quite a bit.